scripts/release.py

```diff
@@ -651,7 +651,11 @@ def prepare(self, dodeploy=False, doperform=False, dryrun=False, upgrade_only=Fa
                 # Maintenance branches are kept, so update their versions
                 log("\n[INFO] Maintenance branch...")
                 msg_commit = "Update %s to %s" % (self.tag, self.maintenance_version)
-                self.update_versions(self.tag, self.maintenance_version)
+                self.repo.git_recurse("reset HEAD~", with_optionals=True)
+                self.update_versions(self.snapshot, self.maintenance_version)
+                for other_version in self.other_versions:
+                    if len(other_version) > 0:
+                        self.update_versions(other_version[0], other_version[2])
```

Comment:
Nice catch! but it will be `self.update_versions(other_version[0], other_version[1])`
The other versions stay aligned on a release (`other_version[1]`) on the maintenance branch.